From a498c256a5abb45437c5efcaf2f0c9f56c645249 Mon Sep 17 00:00:00 2001
From: Jacob Moroni <jmoroni@google.com>
Date: Thu, 18 Dec 2025 23:12:42 +0000
Subject: [PATCH 6/8] irdma: Change default shared credits value to 64

Changed the default value for IRDMA_UD_SHARED_CREDITS to 64. Also
changed credit return to use a normal atomic store.

Signed-off-by: Jacob Moroni <jmoroni@google.com>
---
 providers/irdma/umain.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/providers/irdma/umain.c b/providers/irdma/umain.c
index 457a8f96..d724cbb0 100644
--- a/providers/irdma/umain.c
+++ b/providers/irdma/umain.c
@@ -55,7 +55,7 @@ struct irdma_env_params env = {
 	.ud_qd_override = 0,
 	.transparent_ud_qd_override = 0,
 	.cq_size_override = 0,
-	.shared_ud_credits = 0,
+	.shared_ud_credits = 64,
 };
 
 unsigned int irdma_dbg;
@@ -252,12 +252,9 @@ int try_acquire_credit(uint32_t *cookie)
 void release_credit(int index, uint32_t cookie)
 {
 	struct shared_ud_shm *inst = g_shared_ud_shm;
+	atomic_store_explicit(&inst->acquire_time[index], 0,
+			      memory_order_relaxed);
 
-	/* No need to check init because returning a credit implies it. */
-	atomic_compare_exchange_strong_explicit(&inst->acquire_time[index],
-						&cookie, 0,
-						memory_order_relaxed,
-						memory_order_relaxed);
 }
 
 /* Initialize the backpressure shared memory region for the first time. */
-- 
2.56.0.212.g129ede20d9-goog

